// admin-bro/lib/frontend/components/design-system/badge.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const colorVariant = (0, _styledSystem.variant)({
  variants: {
    primary: {
      color: 'white',
      bg: 'primary',
      'border-color': 'transparent'
    },
    danger: {
      color: 'white',
      bg: 'danger',
      'border-color': 'transparent'
    }
  }
});

const Badge = _styledComponents.default.span.withConfig({
  displayName: "badge__Badge",
  componentId: "c1isbi-0"
})(["border-radius:4px;border:2px solid ", ";", ";", ";", ";", ";"], ({
  theme
}) => theme.colors.primary, _styledSystem.space, _styledSystem.color, _styledSystem.fontSize, colorVariant);

Badge.defaultProps = {
  px: 3,
  py: 2,
  fontSize: 1,
  bg: 'white',
  color: 'primary'
};
var _default = Badge;
exports.default = _default;
