// admin-bro/lib/frontend/components/routes/design-system.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _button = _interopRequireDefault(require("../design-system/button"));

var _wrapperBox = _interopRequireDefault(require("../ui/wrapper-box"));

var _label = _interopRequireDefault(require("../design-system/label"));

var _input = _interopRequireDefault(require("../design-system/input"));

var _textArea = _interopRequireDefault(require("../design-system/text-area"));

var _checkBox = _interopRequireDefault(require("../design-system/check-box"));

var _badge = _interopRequireDefault(require("../design-system/badge"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const DesignSystem = () => {
  const [c1Checked, setC1Checked] = (0, _react.useState)(false);
  return _react.default.createElement(_wrapperBox.default, null, _react.default.createElement(_wrapperBox.default, {
    border: true
  }, _react.default.createElement("h3", {
    style: {
      marginTop: 100
    }
  }, "Buttons"), _react.default.createElement("p", null, _react.default.createElement(_button.default, null, "Regular button"), _react.default.createElement(_button.default, {
    ml: 3,
    variant: "primary"
  }, "Primary"), _react.default.createElement(_button.default, {
    ml: 3,
    variant: "danger"
  }, "Danger"), _react.default.createElement(_button.default, {
    ml: 3,
    variant: "text",
    as: "a"
  }, "Text Link")), _react.default.createElement("p", null, _react.default.createElement(_button.default, {
    size: "sm"
  }, "Small"), _react.default.createElement(_button.default, {
    ml: 3
  }, "Regular size"), _react.default.createElement(_button.default, {
    size: "lg",
    ml: 3
  }, "Large")), _react.default.createElement("h3", null, "Inputs"), _react.default.createElement("p", null, _react.default.createElement(_label.default, null, "This is a label"), _react.default.createElement(_input.default, {
    name: "example1"
  })), _react.default.createElement("p", null, _react.default.createElement(_label.default, {
    variant: "required"
  }, "This is a Required Text Area"), _react.default.createElement(_textArea.default, {
    name: "example2"
  })), _react.default.createElement("p", null, _react.default.createElement(_checkBox.default, {
    id: "example3",
    checked: c1Checked,
    onChange: () => {
      setC1Checked(!c1Checked);
      console.log(c1Checked);
    }
  }), _react.default.createElement(_label.default, {
    htmlFor: "example3",
    variant: "required",
    display: "inline",
    ml: "2"
  }, "This is a checkbox")), _react.default.createElement("h3", null, "Badges"), _react.default.createElement("p", null, _react.default.createElement(_badge.default, null, "default badge"), _react.default.createElement(_badge.default, {
    ml: 3,
    variant: "primary"
  }, "primary badge"), _react.default.createElement(_badge.default, {
    ml: 3,
    variant: "danger"
  }, "danger badge"))));
};

var _default = DesignSystem;
exports.default = _default;
